### DISCUSSION TITLE
請問lab2中的Stream_t是甚麼?

### DISCUSSION THREAD

**Discussion by lumusen0305** [bol-edu/HLS-SOC-Discussions#206] [Q&A]
![stream_t](https://github.com/bol-edu/HLS-SOC-Discussions/assets/45762207/de30025b-07fc-44f1-ba08-86feab9086cb)
請問在lab2的work_books的ppt中出現的Stream_t是我們自己要宣告嗎?還是有lib自帶。

**Comment by roy-tsai-tw** [comment]
依照之前TA的說明和我目前做的狀態，這個Stream_t應該是自定義的。並且要注意裡面的名稱應與EdgeDetect_tb.cpp中描述的相同。

**Comment by ZheChen-Bill** [comment]
stream_t會需要自己設定，
這個stream_t我自己是設定成一個struct，
![圖片](https://github.com/bol-edu/HLS-SOC-Discussions/assets/88698677/70cb3a76-4b2d-4715-97f7-c8ee260e00dd)
你可以在testbench中找到如何使用此data type，
![圖片](https://github.com/bol-edu/HLS-SOC-Discussions/assets/88698677/bbc7d78e-f9c6-4837-82a0-4bebe2dbab8c)
![圖片](https://github.com/bol-edu/HLS-SOC-Discussions/assets/88698677/aa274dcb-00ed-45bc-a468-a106c007a9cf)
如果你想對其做調整也可以，但要記得調整testbench中如何給定input data的。
